src/base/path.cpp

```diff
@@ -155,7 +155,7 @@ QString Path::extension() const
 
 bool Path::hasExtension(const QString &ext) const
 {
-    return (extension().compare(ext, Qt::CaseInsensitive) == 0);
+    return m_pathStr.endsWith((ext.startsWith(QLatin1Char('.')) ? ext : (QLatin1Char('.') + ext)), Qt::CaseInsensitive);
```

Comment:
1. Can you enforce that ext should always start with '.'?
2. Maybe use `u'.'`?
3. Wouldn't case sensitivity differ based on OS?

Reply:
> 3\. Wouldn't case sensitivity differ based on OS?

IIRC, ".ext", ".EXT", ".Ext" etc. are considered as the same extension on all platforms.

Reply:
> 1. Can you enforce that ext should always start with '.' (make a requirement)?

I would make it and add assertion here. `Path::extension()` returns extensions with initial dot so they will be consistent.